mewarpx/mewarpx/assemblies.py

```diff
@@ -25,15 +24,14 @@ class Assembly(object):
     # This is overridden if a diagnostic is installed to record scraped
     # current.
     scraper_diag = None
+    # List of diagnostic functions to call after collecting scraped particles
+    scraper_diag_fnlist = None
 
     # fields is used by the diags.FluxInjectorDiag to know what to write to
     # the CSV file. It can be overridden by child classes, but is not currently
     # adjustable by the user.
 
-    # IF CHANGING THIS, CHANGE IN self.record_scrapedparticles() AS WELL.
-    fields = ['t', 'step', 'species_id', 'V_e', 'n', 'q', 'E_total']
-
-    def __init__(self, V, T, WF, name):
+    def __init__(self, V, T, WF, name, read_scraped_particles):
```

Comment:
I'm not sure about the `read_scraped_particles` flag. Might be cleaner to just automatically have it be `True` for all assemblies and instead control scraping through the C++ flags alone.